src/Microsoft.OpenApi.Readers/V2/OpenApiDocumentDeserializer.cs

```diff
@@ -153,12 +153,23 @@ private static void MakeServers(IList<OpenApiServer> servers, ParsingContext con
                     {
                         host = "//" + host;  // The double slash prefix creates a relative url where the scheme is defined by the BaseUrl
                     }
-
+                    int? port = null;
```

Comment:
>int? [](start = 20, length = 4)

nullable not needed --  int.Parse(pieces.Last()) throws an error if pieces.Last() is not parsable.

Reply:
@PerthCharern I think my logic is correct here.  If the host does not contain ":" then we won't try to call int.Parse(pieces.Last()) and therefore port will be null.  I don't want to update uriBuilder.Port if port isn't specified because the URL should omit port.
If the host does contain ":" and the part after the colon is not a number, I'm ok with it throwing a parsing exception.

Reply:
That makes sense.